python/feathub/feature_tables/tests/test_redis_source_sink.py

```diff
@@ -28,90 +34,140 @@
+                    i + 1, Int64
+                ),
+                b"__timestamp__": int(
+                    to_unix_timestamp(
+                        datetime(2022, 1, 1, 0, 0, i),
+                        format="%Y-%m-%d %H:%M:%S",
+                    )
+                ).to_bytes(8, byteorder="big"),
+            },
+            redis_client.hgetall(key.decode("utf-8")),
         )
 
-        host, port, redis_client = self._get_redis_host_port_and_client()
-
-        sink = RedisSink(
-            namespace="test_namespace",
-            host=host,
-            port=port,
+    redis_client.close()
+
+
+class RedisSourceSinkStandaloneModeITTest(ABC, FeathubITTestBase):
+    def setUp(self):
+        super().setUp()
+        self.redis_container = RedisContainer()
+        self.redis_container.start()
+
+    def tearDown(self):
+        super().tearDown()
+        self.redis_container.stop()
+
+    def test_redis_sink_standalone_mode(self):
+        _test_redis_sink(
+            self,
+            "127.0.0.1",
+            int(
+                self.redis_container.get_exposed_port(
+                    self.redis_container.port_to_expose
+                )
+            ),
+            "standalone",
+            self.redis_container.get_client(),
         )
 
-        self.client.materialize_features(
-            features=source,
-            sink=sink,
-            allow_overwrite=True,
-        ).wait(30000)
 
-        self.assertEquals(len(redis_client.keys("*")), input_data.shape[0])
-
-        for i in range(input_data.shape[0]):
-            key = b"test_namespace:" + serialize_object_with_protobuf(
-                input_data["id"][i], Int64
-            )
-
-            self.assertEquals(
-                {
-                    int(0).to_bytes(4, byteorder="big"): serialize_object_with_protobuf(
-                        i + 1, Int64
-                    ),
-                    b"__timestamp__": int(
-                        to_unix_timestamp(
-                            datetime(2022, 1, 1, 0, 0, i),
-                            format="%Y-%m-%d %H:%M:%S",
-                        )
-                    ).to_bytes(8, byteorder="big"),
-                },
-                redis_client.hgetall(key.decode("utf-8")),
-            )
-
-    @classmethod
-    def _get_redis_host_port_and_client(cls):
-        host = cls.redis_container.get_container_host_ip()
-        if host == "localhost":
-            host = "127.0.0.1"
-        port = cls.redis_container.get_exposed_port(cls.redis_container.port_to_expose)
-        client = cls.redis_container.get_client()
-        return host, int(port), client
+class RedisSourceSinkClusterModeITTest(ABC, FeathubITTestBase):
+    def setUp(self):
+        super().setUp()
+        self.redis_container = RedisClusterContainer()
+        self.redis_container.start()
+        sleep(10)  # wait for container to reach a stable status
```

Comment:
Instead of sleeping 10 seconds here, could we implement the `_wait_container_ready` method? Similar to `RedisContainer`?